src/webgpu/api/validation/capability_checks/limits/limit_utils.ts

```diff
@@ -700,41 +700,62 @@ export class LimitTestsImpl extends GPUTestBase {
     }
   }
 
+  _createRenderPipelineDescriptor(module: GPUShaderModule): GPURenderPipelineDescriptor {
+    return {
+      layout: 'auto',
+      vertex: {
+        module,
+        entryPoint: 'mainVS',
+      },
+    };
+  }
+
+  _createRenderPipelineDescriptorWithFragmentShader(
+    module: GPUShaderModule
+  ): GPURenderPipelineDescriptor {
+    return {
+      layout: 'auto',
+      vertex: {
+        module,
+        entryPoint: 'mainVS',
+      },
+      fragment: {
+        module,
+        entryPoint: 'mainFS',
+        targets: [],
```

Comment:
Using depth-stencil attachment instead is a specific requirement for the `createPipeline, at_over` tests in maxStorageTexturesPerShaderStage.spec.ts due to the extra limit `maxFragmentCombinedOutputResources`, we‘d better just decouple this case from the extra limit.

I also thought about doing the changes in limit_utils.ts, but I didn't know how to explain why we need to enable the opitional depth -stencil attachment and empty color attachmens in the fragment stage for common cases, most cases have nothing to do with depth-stencil attachment. 

After the  #2471, I will also add `create`Compute/Render`PipelineDescriptor` helpers and simply the `createPipeline` function using `createComputePipeline`  and `createRenderPipeline` in the limit_utils.ts, currently the  descriptor statements are redundant in `createPipeline` and `createPipelineAsync`, but it's an optimization, not related to the test issues, to make your review easier, I did not put the changes in the fix PR.

Reply:
> Using depth-stencil attachment instead is a specific requirement for the createPipeline, at_over tests in maxStorageTexturesPerShaderStage.spec.ts due to the extra limit

The requirement not to use a depthStencilAttachment. The requirement is to NOT use a colorAttachment. None of the other tests require a colorAttachment either, They just require a pipeline that succeeds. So, removing the colorAtthacments and using a depthStencilAttachment works for all tests.